### ISSUE TITLE
No preview for meduza.io articles

### ISSUE THREAD

**Issue by rkfg** [matrix-org/synapse#2853]
### Description

There's no preview image for any article from meduza.io news site.

### Steps to reproduce

- post an url like https://meduza.io/slides/vokrug-shum-rekordnyy-snegopad-v-moskve
- only the title and text is shown in the preview without the image

Server drops a line in the log:

2018-02-06 15:31:22,234 - synapse.rest.media.v1.preview_url_resource - 238 - WARNING - GET-12191- Couldn't get dims for https://meduza.io/imgly/share/1517844117/slides/vokrug-shum-rekordnyy-snegopad-v-moskve

The image itself is quite fine, it's a PNG of size 1200x630.

### Version information

- **Homeserver**: own homeserver
- **Version**:  Synapse/0.26.0
- **Install method**: package manager
- **Platform**: set up on Debian 9.3 amd64.

**Comment by neilisfragile** [comment]
$ curl https://meduza.io/imgly/share/1517844117/slides/vokrug-shum-rekordnyy-snegopad-v-moskve > image
$ identify image
image JPEG 1200x630 1200x630+0+0 8-bit sRGB 126632B 0.000u 0:00.000

Not clear on first pass why dims cannot be extracted

**Comment by rkfg** [comment]
So for this article above I have a preview image now. But this one's broken (as well as other more recent articles): https://meduza.io/feature/2021/04/07/proekt-rasskazal-o-vtoroy-zhene-ramzana-kadyrova-i-novoy-lichnosti-ego-pervoy-suprugi

The issue is that their server sends content-type as `image/jpeg; charset=utf-8` which is weird but not against the spec I guess. It can be fixed [in this file](https://github.com/matrix-org/synapse/blob/0d87c6bd121d043d5474cc20e352794ff8cf5625/synapse/rest/media/v1/preview_url_resource.py#L514) like this:
```diff
                 if b"Content-Type" in headers:
                     media_type = headers[b"Content-Type"][0].decode("ascii")
+                    scpos = media_type.find(';')
+                    if scpos > 0:
+                        media_type = media_type[:scpos]
                 else:
                     media_type = "application/octet-stream"
```

**Comment by clokep** [comment]
@rkfg Any chance you'd be willing to put up a PR? `tests/rest/media/v1/test_url_preview.py` has some tests which could be updated too.

**Comment by rkfg** [comment]
I'll see what I can do. There's another issue that prevents the preview images for `pikabu.ru` from showing, around [this line](https://github.com/matrix-org/synapse/blob/0d87c6bd121d043d5474cc20e352794ff8cf5625/synapse/config/repository.py#L73):
```diff
         jpeg_thumbnail = ThumbnailRequirement(width, height, method, "image/jpeg")
         png_thumbnail = ThumbnailRequirement(width, height, method, "image/png")
         requirements.setdefault("image/jpeg", []).append(jpeg_thumbnail)
+        requirements.setdefault("image/jpg", []).append(jpeg_thumbnail)
         requirements.setdefault("image/webp", []).append(jpeg_thumbnail)
         requirements.setdefault("image/gif", []).append(png_thumbnail)
         requirements.setdefault("image/png", []).append(png_thumbnail)
```
Their content-type for the previews is `image/jpg` which seems not to be standard conforming but the Internet is a mess and we have to deal with it. Should I also add this change to the PR?

**Comment by clokep** [comment]
It seems related enough to more robust handling of the `Content-Type` header, so sure.

**Comment by rkfg** [comment]
Opened a PR #9788 that should fix this issue.

**Comment by clokep** [comment]
Fixed via #9788.
